Battery.cpp

```diff
@@ -0,0 +1,84 @@
+#include "Battery.h"
+#include "loggingServices.h"
+
+Battery::Battery() 
+{
+    m_parameterValueMap = {{batteryParamType::Temperature,   0.0},
+                           {batteryParamType::StateOfCharge, 0.0},
+                           {batteryParamType::ChargeRate,    0.0}
+    };
+    m_parameterWarningLevelMap = {{batteryParamType::Temperature,   warningLevelType::disabled},
+                                  {batteryParamType::StateOfCharge, warningLevelType::disabled},
+                                  {batteryParamType::ChargeRate,    warningLevelType::disabled}
+    };
+}
+
+Battery::Battery(float tempValue, warningLevelType tempWarningLevel,
+        float stateOfChargeValue, warningLevelType stateOfChargeWarningLevel,
+        float chargeRateValue, warningLevelType chargeRateWarningLevel )
+        {
+            m_parameterValueMap = {{batteryParamType::Temperature,   tempValue},
+                                   {batteryParamType::StateOfCharge, stateOfChargeValue},
+                                   {batteryParamType::ChargeRate,    chargeRateValue}
+                                  };
+            m_parameterWarningLevelMap = {{batteryParamType::Temperature,   tempWarningLevel},
+                                          {batteryParamType::StateOfCharge, stateOfChargeWarningLevel},
+                                          {batteryParamType::ChargeRate,    chargeRateWarningLevel}
+                                         };
+        }
+
+bool Battery::isInRange(batteryParamType paramType)
+{
+    bool isInRange = false;
+    float l_paramValue = m_parameterValueMap.at(paramType);
+    warningLevelType l_warningLevel = m_parameterWarningLevelMap.at(paramType);
+
+    if(l_paramValue >= parameterRangeMap.at(paramType).first && 
+       l_paramValue <= parameterRangeMap.at(paramType).second)
+    {
+        isInRange = true;
+        std::cout << logRangeWarning( paramType, l_paramValue, l_warningLevel, calcWarningTolerances(paramType));
+    }
+    else
+    {
+        std::cout << logRangeError(paramType);
+    }
+    return isInRange;
+}
+    
+bool Battery::batteryIsOk()
+{
+        return (isInRange(batteryParamType::Temperature) &&
+        isInRange(batteryParamType::StateOfCharge) &&
+        isInRange(batteryParamType::ChargeRate));
+}
+
+std::pair<float, float> Battery::calcWarningTolerances(batteryParamType paramType)
```

Comment:
Use of a `pair` data-type makes it hard to recognize that the first float is min and the second one is max. Prefer using custom types (like struct), where you can name each element.